vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/main/java/com/vaadin/flow/component/spreadsheet/ItemFilter.java

```diff
@@ -183,7 +183,8 @@ protected void initAllItemsCheckbox() {
     protected void initOptions() {
         filterOptionsProvider = new ListDataProvider<>(filterOptions);
         filterCheckbox = new CheckboxGroup<>();
-        filterCheckbox.setDataProvider(filterOptionsProvider);
+        filterCheckbox.setItems(filterOptionsProvider);
```

Comment:
Using the ItemFilter is somewhat difficult due to it closing occasionally:

https://user-images.githubusercontent.com/1222264/173316969-bf1f7203-dc14-4331-8d4c-46968d3cbf3c.mp4

Reply:
It turns out the issue was from the logic to remove the popup button not sent to the client. 

Changed the whole logic to add/remove button to single calls instead of passing the list of popup button state and looping through them on the client.